ODIN_II/parse_odin_stats.py

```diff
@@ -0,0 +1,948 @@
+        , "fanin_outputs_total"
+        , "fanin_outputs_avg"
+        , "num_nodes_to_output"
+        , "max_fanout_node"
+        , "max_fanin_node"
+        , "max_level"
+        , "gnd_path_length"
+        , "vcc_path_length"
+        , "vcc_max_path"
+        , "vcc_min_path"
+        , "max_dist_out"
+        , "min_dist_out"
+        , "top_input_avg_dist_min"
+        , "top_input_avg_dist_max"
+        #, "longest_toplvl_path"
+        #, "shortest_toplvl_path"
+        ,"shortest_maxpath"
+        , "longest_minpath"
+        , "top_lvl_avg_max"
+        , "avg_node_max_dist"
+        , "avg_toplvl_chain_max_length"
+        , "longest_latch_chain"
+        , "longest_consecutive_latch_chain"
+        , "num_edges_fo"
+        , "num_edges_fi"
+        #######################
+        , "vcc_fanout_live"
+        , "gnd_fanout_live"
+        , "pad_fanout_live"
+        , "gnd_fanout_all"
+        , "vcc_fanout_all"
+        , "pad_fanout_all"
+        , "max_input_pathrange"
+        , "min_input_pathrange"
+        , "output_file"
+        ########################## from prev parser
+        , "simulation_time"
+        , "elapsed_time"
+        , "percent_coverage"
+        , "exit_code"
+        , "total_time"
+        , "max_rss"
+        , "context_switches"
+        , "cpu_migration"
+        , "page_faults"
+        , "stalled_cycle_frontend"
+        , "stalled_cycle_backend"
+        , "cycles"
+        , "branches"
+        , "branch_misses"
+        , "llc_loads"
+        , "llc_load_miss"
+        , "percent_cpu_usage"
+        , "minor_page_faults"
+    ]
+        #removed for now - unimplemented
+    """ 
+        , "number_of_connections"
+        , "used_threads"
+        , "degree"
+        , "number_of_stages"
+        """
+
+    """
+    # TODO add --help / -h flag and put all formatting options in sep function, call function on error or help
+    # multimode, multiple inputs into csv and do nothing else 
+    # TODO Streamline all this args stuff into something more elegant
+    # Minimum of 3 inputs required
+    # TODO (maybe) Could do diffs on these as well as long as they are bunched in twos but that is perhaps overkill
+    # TODO Maybe add percent changes as a fourth line
+    # this hinges on the x.abc.blif a.odin.blif format generated by run_vtr_flow and supplied in alphabetical order
+    # This is kind of the inverse of how we presently use compare mode, which assumes the 'reduced' log will be compared against input1
+    # NOTE This now by default does not produce flag files for -mc mode (but still flag warnings) unless "--flags explicitly specified 
+    # Also --concise flag disables printing of summary to stdout
+    # TODO Multicompare works fine with 2 inputs and so compareMode is somewhat redundant, only different in defaulting to flags on
+    # Another way to do multicompare would be to test every line against every previous line, not sure if that is better or worse than using pairs"""
```

Comment:
keep your multiline """ markers location consistent

Reply:
I would add the TODO to the top bellow the help text since it is insightful to know ahead of time